src/test/kotlin/no/nav/tilleggsstonader/sak/vedtak/OpphørValidatorServiceTest.kt

```diff
@@ -0,0 +1,128 @@
+package no.nav.tilleggsstonader.sak.vedtak
+
+import io.mockk.every
+import io.mockk.mockk
+import no.nav.tilleggsstonader.libs.utils.osloDateNow
+import no.nav.tilleggsstonader.sak.behandling.domain.BehandlingType
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.ApiFeil
+import no.nav.tilleggsstonader.sak.util.saksbehandling
+import no.nav.tilleggsstonader.sak.util.tilSisteDagIMåneden
+import no.nav.tilleggsstonader.sak.util.vilkår
+import no.nav.tilleggsstonader.sak.vedtak.barnetilsyn.TilsynBarnTestUtil.vedtakBeregningsresultat
+import no.nav.tilleggsstonader.sak.vedtak.barnetilsyn.beregning.TilsynBarnBeregningService
+import no.nav.tilleggsstonader.sak.vedtak.barnetilsyn.domain.BeregningsresultatTilsynBarn
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.VilkårService
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.VilkårStatus
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.VilkårType
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.Vilkårsresultat
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.VilkårperiodeService
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.VilkårperiodeTestUtil
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.Vilkårperioder
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.felles.Vilkårstatus
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+
+class OpphørValidatorServiceTest {
```

Comment:
Det var litt vanskelig å lese disse testene. Kunne de kanskje vært gruppert litt med nested?

- Korrekt data 
- Validering av nye oppfylte perioder (samling)
- Validering av nye perioder som ikke er oppfylte (ny)
- Beregning
- Validering av tom på endrede perioder (samling)

Reply:
For å gjøre det mer lesbart kunne man kanskje trukket ut noen av de mest brukte ut som variabler. F.eks. perioder som er "innafor", så blir det "return gyldig periode" og enklere å se hva som faktisk er meningen å se på i testen

Reply:
Det ble veldig mye ryddigere. Fin forslag.
5943369e4a805a8bfd0e965a2ef46b2162bed8c5